src/js/components/ToggleCheckboxList/ToggleCheckboxList.tsx

```diff
@@ -0,0 +1,207 @@
+import { PanelHeader } from '../PanelHeader';
+import { PanelBody } from '../PanelBody';
+import { PanelRow } from '../PanelRow';
+import { Tag } from '../Tag';
+import { Field } from '../Field';
+import { ListToggle } from '../ListToggle';
+
+type Props = {};
+
+const ToggleCheckboxList:FunctionComponent<Props> = () => {
+  const [activeRows, setActiveRows] = useState([]);
+  const [headerStatus, setHeaderStatus] = useState({});
+
+  const programData = [{
+    "productName": "Competitive",
+    "seasonName": "2021 Fall Season",
+    "divisions": [
+      "Unassigned",
+      "BoysU14",
+      "BoysU12",
+      "GirlsU14",
+      "GirlsU12"
+    ]
+  }, {
+    "productName": "Recreational",
+    "seasonName": "2021 Fall Season",
+    "divisions": [
+      "BoysU14",
+      "BoysU12",
+      "GirlsU14",
+      "GirlsU12"
+    ]
+  }]
+
+  // Dynamically create refs outside of the mapping below for performance reasons
+  const bodyRefs = React.useRef([]);
+  bodyRefs.current = programData.map((_, i) => bodyRefs.current[i] ?? createRef());
+
+  const toggleAllRows = (productName: string, idx: number) => {
+    let rowData= [];
+
+    if (productName in headerStatus && headerStatus[productName]['activeCount'] > 0) {
+      const { status } = headerStatus[productName];
+
+      // Remove rows which are associated with the header's naming convention
+      rowData = activeRows.filter(row => !row.includes(productName))
+
+      // Remove or filter out the header we do not want to keep
+      const newObject = Object.keys(headerStatus)
+        .filter(item => item != productName)
+        .reduce((res, key) => (res[key] = headerStatus[key], res), {});
+
+      setHeaderStatus(newObject);
+    } else {
+      const productChildElements = getChildren(idx);
+
+      const divisions = [...productChildElements].map((division, index) => {
+        return `${productName}-${division.innerText}-${index}`
+      });
+
+      rowData = [
+        ...activeRows,
+        ...divisions
+      ];
+
+      setHeaderStatus({
+        ...headerStatus,
+        [productName]: {
+          'activeCount': divisions.length,
+          'status': 'true'
+        }
+      });
+    }
+
+    setActiveRows(rowData);
+  }
+
+  const getChildren = (idx: number) => bodyRefs.current[idx].current &&
+    bodyRefs.current[idx].current.children[0].childNodes;
+
+  const onRowClick = (productName: string, division: string, parentIdx: number) => {
+    let newActiveList = [];
+    let activeCount = 0;
+    let newStatus = 'false';
+    const children = getChildren(parentIdx);
+
+    // Row already exists, unselect it
+    if (activeRows.includes(division)) {
+      newActiveList = activeRows.filter(item => item != division);
+
+      activeCount = headerStatus[productName] ?
+        headerStatus[productName]['activeCount'] - 1 :
+        headerStatus[productName]['activeCount'];
+
+      if (activeCount > 0 && activeCount < children.length) {
+        newStatus = 'indeterminate';
```

Comment:
nit: use constants here for the state values: 
ex `import { CheckboxStates } from "../../../types";`
`CheckboxStates.INDETERMINATE`